### PR TITLE
Python: refactor server and client generation to allow easier regeneration

### PR DESCRIPTION
**Pull request by serboctor**
For the client generation always overwrite all files.
 For the server generation:
- Add a new directory handlers that contains a file for each api handler. These files are safe to edit and once they exist won't be overwritten.
- All other files will be overwritten even if they exist, it is the user's responsibility to re-add any changes that were done to these files.

Added comments at the top of all the python templates stating whether this file is safe to edit or not.

Fixes https://github.com/Jumpscale/go-raml/issues/393

### CODE REVIEW

File: codegen/python/server.go [hunk 1]
```diff
@@ -36,3 +36,11 @@ func NewServer(kind string, apiDef *raml.APIDefinition, apiDocsDir string,
 		return nil
 	}
 }
+
+func typesDir() string {
+	return "types"
+}
+
+func handlersDir() string {
```
Comment (iwanbk) [inline codegen/python/server.go]: Why not simply make it as constant?

File: codegen/python/server.go [hunk 2]
```diff
@@ -36,3 +36,11 @@ func NewServer(kind string, apiDef *raml.APIDefinition, apiDocsDir string,
 		return nil
 	}
 }
+
+func typesDir() string {
+	return "types"
```
Comment (iwanbk) [inline codegen/python/server.go]: Why not simply make it as constant?

File: codegen/python/resources.go [hunk 3]
```diff
@@ -14,6 +17,41 @@ type pythonResource struct {
...
+				return err
+			}
+		}
+	}
+
+	methods := struct {
+		Methods []serverMethod
+	}{
+		Methods: allMethods,
+	}
```
Comment (iwanbk) [inline codegen/python/resources.go]: we can also use map.
```go
methods := map[string]interface{}{
     "Methods":allMethods,
}
```
Not sure which one is better but it is certainly shorter

File: codegen/python/server.go (hunk 2 above)
Comment (serboctor) [reply codegen/python/server.go]: the one for the jsonschema is a function so I simply followed the same approach.

File: codegen/templates/python/server_resource_api_sanic.tmpl [hunk 4]
```diff
@@ -1,33 +1,16 @@
...
-import json as JSON
+# DO NOT EDIT THIS FILE. This file will be overwritten when re-running go-raml.
 
-from sanic.response import json, text
-import jsonschema
-from jsonschema import Draft4Validator
-
-import os
-dir_path = os.path.dirname(os.path.realpath(__file__))
-{{ range $k, $v := .ReqBodies }}
```
Comment (iwanbk) [inline codegen/templates/python/server_resource_api_sanic.tmpl]: It seems we can also delete the respected Go `ReqBodies` method

File: codegen/python/resources.go (hunk 3 above)
Comment (serboctor) [reply codegen/python/resources.go]: I am not sure either but this indeed looks better, I'll give it a try.

File: codegen/templates/python/server_resource_api_sanic.tmpl (hunk 4 above)
Comment (serboctor) [reply codegen/templates/python/server_resource_api_sanic.tmpl]: Good catch!

**Pull request merged**